syntax = "proto3";
option go_package = ".;proto";
import "google/protobuf/empty.proto";
service  User {
  rpc GetUserList(PageInfo) returns (UserListResponse); //用户列表
  rpc GetUserByMobile(MobileRequest)  returns (UserInfoResponse); //mobile 查询用户
  rpc GetUserById(IdRequest) returns (UserInfoResponse); // 通过id 查询用户
  rpc CreateUser(CreateUserInfo) returns (UserInfoResponse); // 添加用户
  rpc UpdateUser(UpdateUserInfo) returns (google.protobuf.Empty); //更新用户 更新失败了呢，  应该统一个code 码 表示成功与否先，暂时先这样

}



message PageInfo{
  uint32  pn = 1;
  uint32  pSize = 2;
}
message UpdateUserInfo {
  int32 id = 1;
  string nickName = 2;
  uint64 birthDay = 3;
}
message MobileRequest {
  string mobile = 1;
}

message IdRequest {
  int32 id = 1;
}
message UserListResponse{
  int32  total = 1;
  repeated UserInfoResponse data = 2;

}
message UserInfoResponse {
  int32 id = 1;
  string passWord = 2;
  string mobile = 3;
  string nickName = 4;
  uint64 birthDay = 5;
  string role = 6;
}

message  CreateUserInfo {
  string nickName = 1;
  string passWord = 2;
  string mobile = 3;
}

